### PR TITLE
Feat/profile image upload

### PR DESCRIPTION
**Pull request by jphinning**
- Just made sure every file that is updated in an S3 bucket is deleted first, so everything is synchronized
- Also made sure every table that depends on the User avatar image also has a left join with the Public fle
- Added AWS credentials to the CI/CD pipeline

### CODE REVIEW

File: src/modules/auth/auth.controller.ts [hunk 1]
```diff
@@ -76,9 +76,9 @@ export class AuthController {
     return this.authService.restorePassword(dto);
   }
 
-  @Post('avatar')
   @UseGuards(JwtAuthGuard)
-  @UseInterceptors(FileInterceptor('file'))
+  @Post('avatar')
+  @UseInterceptors(FileInterceptor('avatar'))
```
Comment (DavidSurkov) [inline src/modules/auth/auth.controller.ts]: So interceptor accepts all file types?

Comment (jphinning) [reply src/modules/auth/auth.controller.ts]: Added the multer avatar options, now it only accepts jpeg, png, and jpg as it is in the frontend. Also added the size limitation of 5MB to it

```typescript
@UseInterceptors(FileInterceptor('avatar', multerAvatarOptions))


export const multerAvatarOptions: MulterOptions = {
  fileFilter(
    req: Request,
    file: Express.Multer.File,
    done: (error: Error, acceptFile: boolean) => void,
  ) {
    if (!file.mimetype.match(/\/(jpeg|jpg|png)$/)) {
      done(
        new HttpException('Invalid file type', HttpStatus.BAD_REQUEST),
        false,
      );
    }

    if (file.size > 5e6) {
      done(
        new BadRequestException(
          `The maximum file size has been exceeded. Please select a file smaller than 5MB`,
        ),
        null,
      );
    }

    done(null, true);
  },
};
```

**Review by DavidSurkov — APPROVED** [review]

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -195,13 +195,38 @@ export class AuthService {
     imageBuffer: Buffer,
     filename: string,
   ): Promise<PublicFile> {
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts:192]: but you can get user from jwt validation middleware, why you need additional query?

Comment (jphinning) [reply src/modules/auth/auth.service.ts:192]: Made some changes so the JWT attaches all user information to the request object and removed the first find query. The first update query syncs the database with the S3 bucket and also removes the foreign key if there are any. The other uploads it to the cloud and updates the db with the new image.

**Review by lyudad — APPROVED** [review]

**Pull request merged**